x-pack/legacy/plugins/epm/server/lib/agent/agent.ts

```diff
@@ -5,14 +5,14 @@
  */
 
 import Handlebars from 'handlebars';
-import { VarsEntry } from '../../../common/types';
+import { Dataset } from '../../../common/types';
 
 /**
  * This takes a dataset object as input and merges it with the input template.
  * It returns the resolved template as a string.
  */
-export function createInput(vars: VarsEntry[], inputTemplate: string): string {
-  const view: Record<VarsEntry['name'], VarsEntry['default']> = {};
+export function createInput(vars: Dataset['vars'] = [{}], inputTemplate: string): string {
+  const view: Record<string, string> = {};
```

Comment:
This one is different from the others.

The existing EPM `VarsEntry` defines a type with `name` & `default` properties. However, the [EPR type](https://github.com/elastic/package-registry/blob/master/util/dataset.go#L19) and [golang code](https://github.com/elastic/package-registry/blob/master/util/dataset.go#L19) only say that it's an object. Essentially a `Record<string, any>`

Reply:
@ruflin mind giving a 👍 or 👎 on this ?

Reply:
I wonder if there was a reason I made it `interface` instead of `string`, meaning for some cases we need it. Suggestion:

* Lets keep it in sync with the registry 
* If we change it to `string`, lets do a follow up to both repos to make it string, then we also see th effects.

Reply:
These are in sync now. That's the change.

EPR had a VarsEntry type which was object with keys of default and name. Both values were strings.

However, EPR has vars as an array of interfaces. Essentially an array of plain objects in JS.

I don't see anything in EPR guaranteeing there will be both default and name in the response or that they must be strings.

EPM probably adds these keys/values somewhere but that makes them an EPM type, not an EPR one.

I could add those in EPM, but I wondered if they were really needed since the type seemed so open. i.e. there could be any string value as a key and anything for the value.

Reply:
Lets go forward with this change. 👍